design-doc.md

```diff
@@ -0,0 +1,137 @@
+# Design Document: Adding Support for Next Link Paging with Re-injected Query Parameters
+
+## Overview
+
+Currently, the client treats the `nextLink` in paging as opaque and does not inject query parameters that were part of the initial request when following the next link. This proposal aims to enhance the client by adding capability to reinject query parameters into next requests for paging operations.
+
+## Problem
+
+This is an existing issue in 1-2 brownfield Azure services: Keyvault and potentially AppConfig.
```

Comment:
I guarantee it's in AppConfig :). And in Storage. Storage has two problems: continuation token and query parameters.